### PR TITLE
Feature/memory monitoring

### PR DESCRIPTION
**Pull request by cshcomcom**
## Description
- Now, we can get memory info

### CODE REVIEW

File: util/monitoring.js [hunk 1]
```diff
@@ -83,10 +107,14 @@ async function getMonitoringInfoFromGoogleCloud(projectId, instanceName, keyFile
...
   const info = {};
-  info.cpuUsage = await getCpuUsageInfo(monitoringClient, projectId, instanceName, startTime, endTime);
+  info.cpu = {};
+  info.cpu.usage = await getCpuUsageInfo(monitoringClient, projectId, instanceName, startTime, endTime);
   info.network = {};
   info.network.incoming = await getNetworkReceivedInfo(monitoringClient, projectId, instanceName, startTime, endTime);
   info.network.outgoing = await getNetworkSentInfo(monitoringClient, projectId, instanceName, startTime, endTime);
+  info.memory = {};
+  info.memory.usage = await getMemoryUsageInfo(monitoringClient, projectId, instanceName, startTime, endTime);
+  info.memory.used = await getMemoryBytesInfo(monitoringClient, projectId, instanceName, startTime, endTime);
```
Comment (liayoo) [inline util/monitoring.js:117]: info.memory.bytes ?

File: util/monitoring.js [hunk 2]
```diff
@@ -75,6 +75,30 @@ async function getNetworkReceivedInfo(client, projectId, instanceName, startTime
   return requestAndAssembleInfo(client, request);
 }
 
+async function getMemoryUsageInfo(client, projectId, instanceName, startTime, endTime) {
```
Comment (liayoo) [inline util/monitoring.js:78]: Do we not need instanceName for these queries? If so, how about removing the params from `getMemoryUsageInfo` and `getMemoryBytesInfo`?

Comment (platfowner) [inline util/monitoring.js:78]: What about using 'percent' in the function name (e.g. 'getMemoryPercentInfo()') following google API's original terms ('percent_used')?

Comment (platfowner) [reply util/monitoring.js:78]: And 'utilization' for cpu (e.g. getCpuUtilizationInfo())?

File: util/monitoring.js [hunk 3]
```diff
@@ -83,10 +107,14 @@ async function getMonitoringInfoFromGoogleCloud(projectId, instanceName, keyFile
   }
   const monitoringClient = new googleMonitoring.MetricServiceClient({keyFilename});
   const info = {};
-  info.cpuUsage = await getCpuUsageInfo(monitoringClient, projectId, instanceName, startTime, endTime);
+  info.cpu = {};
+  info.cpu.usage = await getCpuUsageInfo(monitoringClient, projectId, instanceName, startTime, endTime);
```
Comment (platfowner) [inline util/monitoring.js:111]: what about 'info.cpu.utilization'?

File: util/monitoring.js [hunk 4]
```diff
@@ -83,10 +107,14 @@ async function getMonitoringInfoFromGoogleCloud(projectId, instanceName, keyFile
...
   info.network = {};
   info.network.incoming = await getNetworkReceivedInfo(monitoringClient, projectId, instanceName, startTime, endTime);
```
Comment (platfowner) [inline util/monitoring.js:113]: info.network.sent_bytes?

File: util/monitoring.js [hunk 5]
```diff
@@ -83,10 +107,14 @@ async function getMonitoringInfoFromGoogleCloud(projectId, instanceName, keyFile
...
   info.network.outgoing = await getNetworkSentInfo(monitoringClient, projectId, instanceName, startTime, endTime);
```
Comment (platfowner) [inline util/monitoring.js:114]: info.network.received_bytes?

File: util/monitoring.js [hunk 6]
```diff
@@ -83,10 +107,14 @@ async function getMonitoringInfoFromGoogleCloud(projectId, instanceName, keyFile
...
+  info.memory = {};
+  info.memory.usage = await getMemoryUsageInfo(monitoringClient, projectId, instanceName, startTime, endTime);
```
Comment (platfowner) [inline util/monitoring.js:116]: info.memory.percent_used?

File: util/monitoring.js (hunk 1 above)
Comment (platfowner) [reply util/monitoring.js:117]: or info.memory.bytes_used?

Comment (cshcomcom) [reply util/monitoring.js]: Thanks done! I followed google filter(memory/bytes_used)!

File: util/monitoring.js (hunk 2 above)
Comment (cshcomcom) [reply util/monitoring.js]: Thanks! done

Comment (cshcomcom) [reply util/monitoring.js]: Looks good! Done

File: util/monitoring.js (hunk 4 above)
Comment (cshcomcom) [reply util/monitoring.js:113]: Done! Since benchmark has been using camel case, I applied camel case.

File: util/monitoring.js (hunk 5 above)
Comment (cshcomcom) [reply util/monitoring.js:114]: Done! ditto

File: util/monitoring.js (hunk 6 above)
Comment (cshcomcom) [reply util/monitoring.js]: Done! ditto

File: util/monitoring.js (hunk 3 above)
Comment (cshcomcom) [reply util/monitoring.js]: Looks good! Done

**Comment by cshcomcom** [comment]
All comments are addressed! PTAL @liayoo @platfowner

**Review by platfowner — APPROVED** [review]
LGTM
Thanks!

**Review by liayoo — APPROVED** [review]
LGMT, thanks!

**Comment by cshcomcom** [comment]
Thanks for reviews! Now merging~

**Pull request merged**